exalearn-ui/src/app/components/assign-test-modal/assign-test-modal.component.ts

```diff
@@ -5,32 +5,47 @@ import { MatSelectChange } from '@angular/material/select';
 import { MatCheckboxChange } from '@angular/material/checkbox';
 import { EnglishLevels } from '../../enums/enums';
 import { UserBack } from '../../interfaces/interfaces';
+import { ApiService } from '../../services/api.service';
+import { FormControl, Validators } from '@angular/forms';
+import { UserService } from '../../services/user.service';
 
 @Component({
 	selector: 'app-assign-test-modal',
 	templateUrl: './assign-test-modal.component.html',
 	styleUrls: ['./assign-test-modal.component.scss']
 })
 export class AssignTestModalComponent {
-	level: string | null;
-	date: Date | null;
+	level: EnglishLevels | null = null;
+	date: Date | null = null;
 	sendEmail = true;
 	minDate: Date;
-	maxDate: Date;
 	levels = EnglishLevels;
 	DayInMilliseconds = 86400000;
 	levelsValues = Object.values(this.levels);
+	levelControl = new FormControl('', Validators.required);
+	dateControl = new FormControl('', Validators.required);
 
 	constructor(
 		public dialogRef: MatDialogRef<AssignTestModalComponent>,
-		@Inject(MAT_DIALOG_DATA) public data: UserBack
+		@Inject(MAT_DIALOG_DATA) public data: UserBack,
+		public apiService: ApiService,
+		public userService: UserService
 	) {
 		const currentDate = new Date();
 		this.minDate = new Date(currentDate.valueOf() + this.DayInMilliseconds);
 	}
 
 	assign(): void {
-		this.dialogRef.close();
+		if (this.level !== null && this.date !== null && this.data) {
+			this.apiService.postRequest('/api/users/assignedTest', {
```

Comment:
You should use api.constants in postRequest

Reply:
POST_USER_ASSIGN_TEST added and used